File: leetcode/0921-minimum-add-to-make-parentheses-valid/0921-minimum-add-to-make-parentheses-valid.java
class Solution {
    public int minAddToMakeValid(String s) {
        int o = 0, c = 0;
        for (char ch : s.toCharArray()) {
            if (ch == '(') o++;
            else if (o > 0) o--;
            else c++;
        }
        return o + c;
    }
}

File: leetcode/0921-minimum-add-to-make-parentheses-valid/0921-minimum-add-to-make-parentheses-valid.py
class Solution:
    def minAddToMakeValid(self, s: str) -> int:
        pos = 0
        nag = 0
        for c in s:
            if c == '(':
                pos += 1
            elif pos > 0:
                pos -= 1
            else:
                nag += 1

        return pos + nag

        